ZcashFoundation/zebra
# change(ui): Enable the progress bar feature by default, but only show progress bars when the config is enabled

## TODO

After this PR merges, add:
- [x] A changelog feature entry asking for user testing
- [ ] A post on the forums encouraging users to test it after it is released, and mentioning [the known issue with estimates](https://github.com/console-rs/indicatif/issues/556)
- [ ] A mention in the Arborist call

## Motivation

Currently the `progress-bars` feature turns on progress bars and file logging when it is enabled.

To make it easier for users to test progress bars, we can make the progress bars configurable with `zebrad.toml`. This allows us to compile the `progress-bar` feature by default, but not activate it for every user by default.

(Activating progress bars by default could be a breaking change for some users that expect Zebra to write logs to standard output.)

Close #7363
Close #7364

### Complex Code or Requirements

When the progress bar feature is activated, the log file also needs to be activated (unless the user has specifically turned it off).

I added a summary and detailed config, because we might want to have an "expert" mode. But we need to know what users want first!

## Solution

- Add a config for progress bars, which is only used when the `progress-bars` feature is enabled
    - If they are not enabled, log a message telling users how to enable them in the config
    - Document how to enable them in the README
- Enable the `progress-bars` feature by default in Zebra builds

### Testing

- Add a config with progress bars enabled to the config tests
- Make the config tests handle log files correctly
- Check the config path in the config tests

## Review

This is a low priority feature, it doesn't need to go in the 1.3.0 release.

### Reviewer Checklist

  - [ ] Will the PR name make sense to users?
    - [ ] Does it need extra CHANGELOG info? (new features, breaking changes, large changes)
  - [ ] Are the PR labels correct?
  - [ ] Does the code do what the ticket and PR says?
    - [ ] Does it change concurrent code, unsafe code, or consensus rules?
  - [ ] How do you know it works? Does it have tests?

## Follow Up Work

Work out which progress bars users want.
